### DISCUSSION TITLE
✅ Visual indicators for widget display conditions in the structure tree.

### DISCUSSION THREAD

**Discussion by lllopo** [elementor/elementor#28027] [Feature Request]
### Prerequisites

- [X] I have searched for similar feature requests in both open and closed discussions and cannot find an existing request.
- [X] I have verified that the feature is still missing from the latest stable versions of Elementor or Elementor Pro.

### What problem is your feature request going to solve? Please describe..

When a widget has display conditions set on it, it is very hard to find out this. It is buried in selecting the widget and going to its advanced settings tab. A better way to get an overview of this is badly needed as on a page with many conditions that has to be modified it is very easy to miss some without proper overview.

### Describe the solution you'd like.

What I think is the best solution to this is  to add some sort of visual indicator in the structure tree. Like all widgets that have any sort fo display conditions set on them to get a small icon in the tree indicating that. It would be even better if this icon is clickable and takes you straight to the display conditions of this widget.

### Agreement

- [X] I agree that my request may be closed without action if it doesn't meet all the requirements.

**Comment by black-eye** [comment]
The indicator is necessary not only in the structure tree, it would be needed directly in the editor as well. 

Simple icon would do, see how Dynamic Conditions for Elementor by RTO has done this (the little blue eye icon in the top right corner):

![obrazek](https://github.com/user-attachments/assets/681123e9-a1ce-4ada-903f-9b5e15efe0b1)

It's crutial to see which parts of the page are static and which are displayed conditionally, especially when less advanced editors are involved in editing process.

**Reply by nicholaszein** [reply]
@black-eye 

https://elementor.com/blog/elementor-324-floating-bars-variable-fonts-enhanced-css/#:~:text=%5BPro%5D%20Fine%2DTuned%20Display%20Conditions%20for%20Greater%20Control

**Reply by black-eye** [reply]
> @nicholaszein 
> 
> https://elementor.com/blog/elementor-324-floating-bars-variable-fonts-enhanced-css/#:~:text=%5BPro%5D%20Fine%2DTuned%20Display%20Conditions%20for%20Greater%20Control

OK, nice, so in the last version there should be a line and icon in the navigator, that's a step in a good direction. But my point was, that it would be better to have another indicator directly in the editor (in a manner of Dynamic Conditions for Elementor by RTO). Should I create a new request for it, as this request is technically delivered?

BTW, in the navigator, I can see the line, but not the icon. Page refreshed, cache flushed.

**Reply by nicholaszein** [reply]
@black-eye,

we won't be adding an indicator in the canvas, only in the Structure Panel (former Navigator). It's an UI/UX decision. All other indicators such as custom position and custom CSS are in the same place.

**Reply by black-eye** [reply]
@nicholaszein OK, never mind. 

It kind of confirms me, why I have chosen another page builder for new projects. Still, I was hoping Elementor would get better, so maintaining the old sites would get easier. For now, I'll stick with the old Dynamic Conditions for Elementor plugin, simply because it works better for me (I simply need to see what sections are affected by conditions right away on the canvas). One more plugin needed, but on the other hand, I won't have to update the pages for the new Elementor's native conditions.

**Comment by nicholaszein** [comment]
Hello there!

We have great news! 🙌 

### 📢 We're happy to announce that this feature was released with Elementor Pro v3.24.0! 🥳 

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
